app/views/view_components/preview.html.erb

```diff
@@ -1,13 +1,11 @@
-<% if @render_args[:component] %>
-  <% if ViewComponent::Base.render_monkey_patch_enabled || Rails.version.to_f >= 6.1 %>
-    <%= render(@render_args[:component], @render_args[:args], &@render_args[:block]) %>
-  <% else %>
-    <%= render_component(@render_args[:component], &@render_args[:block]) %>
-  <% end %>
+<% @render_args ||= render_args %>
+<% if ViewComponent::Base.render_monkey_patch_enabled || Rails.version.to_f >= 6.1 %>
+  <%= render(@render_args[:component], @render_args[:args], &@render_args[:block]) %>
 <% else %>
   <%= render template: @render_args[:template], locals: @render_args[:locals] || {} %>
 <% end %>
 
-<% if ViewComponent::Base.show_previews_source %>
+<% # Hide the source because it doesn't appear to be significant to system tests %>
```

Comment:
I think we should remove this comment as the `hide_preview_source` param feels more generic than just for this use-case.

```suggestion
```